platform/barefoot/docker-syncd-bfn-rpc.mk

```diff
@@ -17,6 +17,7 @@ SONIC_INSTALL_DOCKER_IMAGES += $(DOCKER_SYNCD_BFN_RPC)
 endif
 
 $(DOCKER_SYNCD_BFN_RPC)_CONTAINER_NAME = syncd
+$(DOCKER_SYNCD_BFN_RPC)_VERSION = 1.0.0
```

Comment:
When multiple users are making updates to syncd code base, who/how/when controls this version?
Even a major change in underlying common libs used could affect version of this docker,

We need a system to enforce before committing this.

Reply:
As discussed, currently the version increment is done by developer in PR. The version is incremented in according to semver rules (breaking change, new functionality introduced, bug fix). When the version changes - ideally on every change related to corresponding component, but to be more realistic let's consider the following. Let's put aside master branch, since regular user does not want to run master branch. It only matters for release branches, but in release branches only bug fixes are allowed, so only last digit changes. It is unusual that a package will depend on particular bug fix, it will rather depends on functionality, thus, a version can be changed once on release branch out. From that moment no new functionality or breaking changes are allowed so all app.ext. packages can still work with this release. 
So, it is Ok if we update major, minor version once a release and then don't update patch version. This is one time operation for every package per release so it is realistic to do. So, I don't think that this cannot be committed before we introduce automatic version increment. Besides, if automatic version bump can be done till next release in phase 2 we even won't be required to do it manually for next release.
Besides, in this particular example, since no one can really depend on syncd, because SAI programming is planned for phase 2 (as well as automatic version update), does it really matter for syncd?

Reply:
Wish it is as simple as you suggest. 
Can you please create a write up on your plan for this versioning that includes when/how/who updates the following
1) Base OS version updates
2) Base OSversion compatibility in each docker
3) Individual docker versions

We would need to pass it around and have a dedicated meeting for this?

Reply:
@renukamanavalan Versioning guidelines were added to HLD https://github.com/Azure/SONiC/blob/b154820484fa297e4cb3a010c8f5709e891edc43/doc/sonic-application-extention/sonic-versioning-strategy.md. and updated based on discussion last week.